zipline/finance/slippage.py

```diff
@@ -175,14 +175,45 @@ def asdict(self):
         return self.__dict__
 
 
-class EquitySlippageModel(SlippageModel):
+class EquityAndFutureMeta(ABCMeta):
+    """
+    This metaclass allows users to create custom slippage models that support
+    both equities and futures by subclassing EquityFutureModel and
+    FutureSlippageModel together.
+
+    Take for example the following custom model:
+
+        class MyCustomSlippage(EquitySlippageModel, FutureSlippageModel):
+            def process_order(self, data, order):
+                ...
+
+    Ordinarily the first parent class in the MRO ('EquitySlippageModel' in this
+    example) would override the 'allowed_asset_types' class attribute to only
+    include equities. However, this is probably not the expected behavior for a
+    reasonable user, so the metaclass will handle this specific case by
+    manually allowing both equities and futures for the class being created.
+    """
+    def __new__(mcls, name, bases, dict_):
+        try:
+            if EquitySlippageModel in bases and FutureSlippageModel in bases:
+                dict_['allowed_asset_types'] = (Equity, Future)
+        except NameError:
```

Comment:
I like this! Only comment I have is that in the `for` loop the word "marker" is slightly confusing in that it's only maybe a marker; might just say "base_class".